Lib/idlelib/idle_test/test_debugger.py

```diff
@@ -23,7 +34,456 @@ def test_init(self):
         debugger.NamespaceViewer(self.root, 'Test')
 
 
-# Other classes are Idb, Debugger, and StackViewer.
+class MockFrameType(object):
+    """Reflection of the types.FrameType."""
+    f_back = None
+    f_builtins = None
+    f_code = None
+    f_globals = None
+    f_lasti = None
+    f_lineno = None
+    f_locals = None
+    f_restricted = 0
+    f_trace = None
+
+    def __init__(self, code, lineno):
+        self.f_code = code
+        self.f_lineno = lineno
+
+
+class IdbTest(unittest.TestCase):
+
+    def test_init_runs_bdb_init(self):
+        """Test that Idb calls the base Bdb __init__."""
+        idb = debugger.Idb(None)
+        assert idb.gui is None
+        assert hasattr(idb, 'breaks')
```

Comment:
These `assert` statements still need to be converted to unittest calls.

Reply:
fixed